homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -86,33 +131,54 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
                             condition,
                             f"{name}{SENSORS[condition][0]} {inverter}",
                             SENSORS[condition][1],
+                            coordinator,
                         )
                     )
-
+                    _LOGGER.debug(
+                        "Adding inverter SN: %s - Type: %s.",
+                        f"{name}{SENSORS[condition][0]} {inverter}",
+                        condition,
+                    )
+            elif "Unable to connect to Envoy" in inverters:
+                _LOGGER.error(
+                    "Unable to connect to Enphase Envoy during setup. Inverter entities not added. Please check IP address and credentials are correct."
+                )
+            elif "can't handle event type ConnectionClosed" in inverters:
+                _LOGGER.error(
+                    "Communication error with Enphase Envoy during setup. Inverter entities not added."
+                )
         else:
             entities.append(
                 Envoy(
                     envoy_reader,
                     condition,
                     f"{name}{SENSORS[condition][0]}",
                     SENSORS[condition][1],
+                    coordinator,
                 )
             )
+            _LOGGER.debug(
+                "Adding sensor: %s - Type: %s.",
+                f"{name}{SENSORS[condition][0]})",
+                condition,
+            )
     async_add_entities(entities)
 
 
 class Envoy(Entity):
-    """Implementation of the Enphase Envoy sensors."""
+    """Envoy entity."""
```

Comment:
Inherited CoordinatorEntity, is this all I have to do?

Reply:
There is a bit more:
* No need to override `should_poll`.
* No need to override `available`.
* No need to override `async_added_to_hass`.
* And you want to transform the code from `async_update` into overriding `state`. (There is a high-level introduction into `CoordinatorEntity` at https://developers.home-assistant.io/docs/integration_fetching_data/)

Reply:
Oh, and you only need to inherit from `CoordinatorEntity`, not from `Entity` anymore.

Reply:
Sorry put the comment in the wrong section! 😮 

- Removed should_poll
- Removed available
- Removed async_added_to_hass
- Removed async_update and moved the code into state
- In state it wasn't a copy and paste from async_update, but mostly the same code exists. Tried to reduce the paths and catch the same errors. I moved the update for the inverter last_reported attribute into the device_state_attributes() method

I noticed now the sensors have values on the first poll rather than waiting for the second poll!